src/Components/App/App.js

```diff
@@ -5,22 +5,72 @@ import Header from '../Header/Header';
 import Home from '../Home/Home';
 import Horoscope from '../Horoscope/Horoscope';
 import Footer from '../Footer/Footer';
+import { fetchHoroscope } from '../../apiCalls';
 
 class App extends Component {
   constructor() {
     super() 
     this.state = {
+      isLoading: true,
+      errorMsg: '',
+      day: '',
+      sign: '',
+      horoscope: '',
+      image: ''
+    }
+  }
+
+  retrieveHoroscopeData = async (url, src, alt, when) => {
+    await fetchHoroscope(url)
+    .then(result => {
+        this.setState({ isLoading: true, horoscope: '' })
+        if (typeof result === 'string') {
```

Comment:
Cool use of settimeout here I think the loading message adds a lot to the vibe and styling of the app!